#45 3 create module

PhilippeLeopoldie opened the pull request.

PhilippeLeopoldie committed:
Add DTO classes for module creation and updates

- Introduced `ModuleCreateDto` for module creation, inheriting from `ModuleForManipulationDto`.
- Added `ModuleDto` as a record type with properties for module details.
- Updated `ModuleForManipulationDto` with validation attributes for required fields.
- Created `ModuleUpdateDto` for updating modules, including an `Id` property.

PhilippeLeopoldie committed:
Refactor Module properties for clarity and flexibility

Renamed properties `Starts` and `Ends` to `StartsAt` and `EndsAt` for consistency with other Entities . Changed the type of `Activities` from `ICollection<Activity>` to `IEnumerable<Activity>` to allow for more flexible, read-only access.

PhilippeLeopoldie committed:
Add module DTO mappings and update profile configurations

- Introduced `LMS.Shared.DTOs.ModuleDtos` for module-related DTOs.
- Implemented bidirectional mapping between `Module` and `ModuleDto`.
- Added custom mapping for `StartsAt` in `ModuleUpdateDto` to `Module`.
- Added custom mapping for `StartsAt` in `ModuleCreateDto` to `Module`.

PhilippeLeopoldie committed:
Refactor ModuleRepository and RepositoryBase classes

Updated ModuleRepository to accept ApplicationDbContext directly in the constructor and removed the previous constructor. Added a Context property in RepositoryBase to enhance access to the database context.

PhilippeLeopoldie committed:
feat(ModuleService): Create Module Service

PhilippeLeopoldie committed:
Merge updated Development branch into current branch

ChornaIryna commented on Domain.Models/Entities/Module.cs:
```diff
@@ -3,12 +3,12 @@ namespace Domain.Models.Entities;
     public int CourseId { get; set; }
 
-    public ICollection<Activity> Activities { get; set; }
+    public IEnumerable<Activity> Activities { get; set; }
```
Varför ändra `IColletion ` till `IEnumerable`?

PhilippeLeopoldie replied:
Jag har använt IEnumerable i mina DTO:er, men nu förstår jag att det är ICollection jag bör använda i mina entiteter, jag har korrigerat, tack!

PhilippeLeopoldie committed:
Change Activities property to ICollection

Updated the Activities property type from IEnumerable<Activity> to ICollection<Activity>. This change enhances the functionality by allowing direct manipulation of the collection, such as adding and removing items.

JorgenHagglund commented:
Varför är vissa Dto'er records och andra classes?

PhilippeLeopoldie committed:
Refactor DTOs to use records for immutability

Updated `ModuleCreateDto`, `ModuleForManipulationDto`, and `ModuleUpdateDto` to records.

PhilippeLeopoldie commented:
> Why are some Dto's records and others classes?

Jag har korrigerat, alla är records nu, tack!

JorgenHagglund reviewed (commented):
Push'a, sedan är det bara att merga, som jag ser det.
